core/templates/dev/head/exploration_editor/ExplorationEditor.js

```diff
@@ -641,9 +641,14 @@ oppia.controller('ExplorationSaveAndPublishButtons', [
         siteAnalyticsService.registerSavePlayableExplorationEvent(
           explorationData.explorationId);
       }
-
       $scope.isSaveInProgress = true;
-      explorationData.save(changeList, commitMessage, function() {
+      explorationData.save(changeList, commitMessage, function(response) {
```

Comment:
I don't think this should be a part of an error callback, since in case of the error callback - response.data won't be defined.. thus it will be indeterminable when to show the version mismatch modal on frontend!

I believe that the line `$log.info('Changes to this exploration were saved successfully.');` should be removed from [here](https://github.com/oppia/oppia/pull/2456/files#diff-078376ddb553fbe57e8abf985fff0e36R137) and instead be placed just outside this [`if condition`](https://github.com/oppia/oppia/pull/2456/files#diff-2792f5ec9e6593c955ffb30beb35fcfeR646) - since if you see the current functionality, then the [`save method`](https://github.com/oppia/oppia/blob/ae23c9b99954b3860d244869d07471975c0674e8/core/templates/dev/head/exploration_editor/EditorServices.js#L129) is being called and inside the `.then( ` function we are immediately logging that the save was successful - without checking for the validity of the draft version (is_version_of_draft_valid, draft_changes) in the data returned by the backend!